src/main/java/frc/robot/commands/MoveIntake.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.Intake.IntakePosition;
+
+
+public class MoveIntake extends CommandBase {
+    
+    private final IntakePosition m_position;
+    private final IntakePosition m_currentIntakePostion = Intake.getCurrentIntakePosition();
+
+    public MoveIntake(IntakePosition position) { 
+        m_position = position;
+        switch (m_position) {
```

Comment:
You don't need to consider the currentIntakePosition in this switch statement. You'll have an encoder out the output shaft of the intake arm. So you don't need to know where the last place you commanded the arm to be is. 

What you want to do is use [PID control](https://docs.wpilib.org/en/latest/docs/software/commandbased/pid-subsystems-commands.html) to move to the position for a certain action. I would recommend making this a PIDCommand. In fact, the motor controllers you will likely use (the TalonSRX) have built-in PID functions, so you don't necessarily have to use PIDCommand (which is why I suggested a moveToPosition function in the subsystem), but I think it's probably easier to understand using PIDCommand.

Reply:
I think that last paragraph is really confusing so I'll try to reword. 

I think you should make MoveIntake a PIDCommand. It would measure the arm's angle using an encoder in the arm's gearbox, and run the motors based on where it wants the arm to be. I can explain this further.